src/python/pants/backend/python/lint/bandit/rules_integration_test.py

```diff
@@ -0,0 +1,117 @@
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.subsystem.util import init_subsystems
+from pants.testutil.test_base import TestBase
+
+
+class BanditIntegrationTest(TestBase):
+  good_source = FileContent(path="test/good.py", content=b"hashlib.sha256()\n")
+  bad_source = FileContent(path="test/bad.py", content=b"hashlib.md5()\n")
+  # MD5 is a insecure hashing function
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(targets={'python_library': PythonLibrary})
+
+  @classmethod
+  def rules(cls):
+    return (
+      *super().rules(),
+      *bandit_rules(),
+      *download_pex_bin.rules(),
+      *pex.rules(),
+      *python_native_code.rules(),
+      *subprocess_environment.rules(),
+      RootRule(CreatePex),
+      RootRule(Bandit),
+      RootRule(BanditTarget),
+      RootRule(PythonSetup),
+      RootRule(PythonNativeCode),
+      RootRule(SubprocessEnvironment),
+    )
+
+  def setUp(self):
+    super().setUp()
+    init_subsystems([Bandit, PythonSetup, PythonNativeCode, SubprocessEnvironment])
+
+  def run_bandit(
+    self,
+    source_files: List[FileContent],
+    *,
+    config: Optional[str] = None,
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.bandit"]
+    if config:
+      # TODO: figure out how to get this file to exist...
+      self.create_file(relpath=".bandit", contents=config)
+      args.append("--bandit-config=.bandit")
+    if passthrough_args:
+      args.append(f"--bandit-args={passthrough_args}")
+    if skip:
+      args.append(f"--bandit-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    target = BanditTarget(
+      PythonTargetAdaptor(
+        sources=EagerFilesetWithSpec('test', {'globs': []}, snapshot=input_snapshot),
+        address=Address.parse("test:target"),
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    result = self.run_bandit([self.good_source])
+    assert result.exit_code == 0
+    assert "No issues identified." in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    result = self.run_bandit([self.bad_source])
+    assert result.exit_code == 1
+    assert "Issue: [B303:blacklist] Use of insecure MD2, MD4, MD5" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    result = self.run_bandit([self.good_source, self.bad_source])
+    assert result.exit_code == 1
+    assert "test/good.py" not in result.stdout
+    assert "Issue: [B303:blacklist] Use of insecure MD2, MD4, MD5" in result.stdout
+
+  def test_respects_config_file(self) -> None:
```

Comment:
I think you'll need to mark this with `@pytest.mark.skip`. I'm probably not going to land the fix for these config tests for a few days and I don't want you to be blocked by that PR.